File: TestsManager/Service/Recording/RecordedEvents/KeypressEvent.cs
namespace Service.Recording.RecordedEvents
{
    public class KeypressEvent : IEvent
    {
        public Guid Id {  get; set; } = Guid.NewGuid();
        public DateTime Timestamp { get; set; } = DateTime.Now;

        public required string Keys { get; set; }


        public IEvent Clone()
        {
            return new KeypressEvent { Id = Id, Timestamp = Timestamp, Keys = Keys };
        }
    }
}

File: TestsManager/Frontend/src/app/models/events/keypress-event.ts
import { IEvent } from './event-interface';

export class KeypressEvent implements IEvent {
    public type: string = 'keypress';
    
    public id: string;
    public timestamp: Date;
    public keys: string;

    constructor(
        id: string,
        timestamp: Date,
        keys: string
    ) {
        this.id = id;
        this.timestamp = timestamp;
        this.keys = keys;
    }
}
